File: example1/src/main/java/com/example/grpc/echo/unary/UnaryBlockingClient.java
package com.example.grpc.echo.unary;

import com.example.grpc.Loggers;
import com.example.grpc.echo.EchoRequest;
import com.example.grpc.echo.EchoServiceGrpc;
import io.grpc.ManagedChannelBuilder;
import io.grpc.StatusRuntimeException;

import java.util.concurrent.TimeUnit;
import java.util.logging.Logger;

public class UnaryBlockingClient {

    private static final Logger logger = Logger.getLogger(UnaryBlockingClient.class.getName());

    public static void main(String[] args) throws Exception {
        Loggers.init();

        var channel = ManagedChannelBuilder.forAddress("localhost", 50051).usePlaintext().build();

        try {
            var blockingStub = EchoServiceGrpc.newBlockingStub(channel);
            var request = EchoRequest.newBuilder().setMessage("world").build();
            var response = blockingStub.unaryEcho(request);
            logger.info("response: " + response.getMessage());
        } catch (StatusRuntimeException e) {
            logger.warning("error: " + e.getStatus());
        } finally {
            channel.shutdown().awaitTermination(30, TimeUnit.SECONDS);
        }
    }
}

File: _kotlin/src/main/java/com/example/grpc/echo/unary/UnaryBlockingClient.kt
package com.example.grpc.echo.unary

import com.example.grpc.EchoRequest

object  /*TODO*/ UnaryBlockingClient {
    private val logger: java.util.logging.Logger =
        java.util.logging.Logger.getLogger(UnaryBlockingClient::class.java.getName())

    @kotlin.Throws(java.lang.Exception::class)
    @kotlin.jvm.JvmStatic
    fun main(args: kotlin.Array<kotlin.String>) {
        Loggers.init()

        val channel: ManagedChannel = ManagedChannelBuilder.forAddress("localhost", 50051).usePlaintext().build()

        try {
            val blockingStub: Unit /* TODO: class org.jetbrains.kotlin.nj2k.types.JKJavaNullPrimitiveType */? =
                EchoServiceGrpc.newBlockingStub(channel)
            val request: Unit /* TODO: class org.jetbrains.kotlin.nj2k.types.JKJavaNullPrimitiveType */? =
                EchoRequest.newBuilder().setMessage("world").build()
            val response: Unit /* TODO: class org.jetbrains.kotlin.nj2k.types.JKJavaNullPrimitiveType */? =
                blockingStub.unaryEcho(request)
            UnaryBlockingClient.logger.log(java.util.logging.Level.INFO, "response: {0}", response.getMessage())
        } catch (e: StatusRuntimeException) {
            UnaryBlockingClient.logger.log(java.util.logging.Level.WARNING, "RPC error: {0}", e.getStatus())
        } finally {
            channel.shutdown().awaitTermination(30, TimeUnit.SECONDS)
        }
    }
}
